client-lib/nsa.js

```diff
@@ -0,0 +1,210 @@
+const deepAssign = require('deep-assign');
+
+const emitter = require('./emitter');
+const formatTime = require('./format-time');
+const sendToAddon = require('./send-to-addon');
+const sendMetricsEvent = require('./send-metrics-event');
+
+const YtCtrl = require('./ctrls/yt-ctrl');
+const AudioCtrl = require('./ctrls/audio-ctrl');
+const VideoCtrl = require('./ctrls/video-ctrl');
+
+let currentStep, playerMap = {}; // eslint-disable-line no-unused-vars
+
+function step() {
+  let currentTime = 0;
+  let progress = window.AppData.progress;
```

Comment:
Suggestion: the `window.AppData.foo` pattern recurs in this file a _lot_. What about wrapping the `window.AppData` proxy object in a module? This change would way simplify unit testing code that touches AppData (just override require to pass in a sinon spy), and this file would become cleaner, too:
```js
const d = require('./app-data');

// instead of let progress = window.AppData.progress, you could do:
let progress = d.progress;
```